Moth cuisine

FluffiestFloof opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Moth Foods, the three C's: cheese, coleslaw, and cotton.

Ported from TG moth cuisine, minus the whole Moth lore, funky names and some extra steps that just felt too bloaty. The original names and description can be re-added, I just felt like it didn't fit.

To be honest it's not really specifically "moth cuisine" as I renamed everything with their 'proper names' so it can just be a bunch of new normal recipe. If you ignore the cotton for some of them.
This includes a lot of new dishes, some incomplete due to a lack of certain ingredients, some adapted.

All dishes with a "*" means they are incomplete from their ss13 counterpart due to missing/placeholder ingredients. I'll try to keep these updated when ingredients eventually gets added for them.
Currently the main placeholders is Ambrosia for Herbs and Ketchup for the Tomato Sauce.

**Currently variously adds:**
- Three new flavors: cotton, vanilla, seeds 
- The cake mail has a chance to come with a Mothmallow 'cake'
- A new mail "Moffin" which contains Moffins
  - Would be more fitting if only moth could receive it but eh, just don't get lucky.

**Adds the following ingredients with recipe:**
- Curd Cheese
- Cheese Curds
- Mozzarella
- Pesto
- Bechamel
- Tomato sauce

**Adds the following dishes with recipe:**
- Base Ingredients
  - Herby Cheese*
  - Mothic Salad
- Entrees
  - Squeaking Stir Fry
  - Toasted Seeds
  - Sweet Chili Cabbage Wrap*
  - Baked Cheese Wheel
  - Baked Cheese Platter
  - Baked Rice + Portion*
- Meals
  - Green Lasagne + Slice
  - Oven-Baked Corn
  - Buttered Baked Corn
  - Mozzarella Sticks
  - Mac Balls*
- Soup and Stews
  - Cotton Soup
  - Cheese Soup
  - Seed Soup
  - Moth Eyeball Soup
  - Bean Stew
  - Oat Stew*
  - Heartburn Soup*
  - Mulan Congee*
  - Cornmeal Porridge*
  - Cheesy Porridge
  - Fried Eggplant and Polenta*
  - Vegetarian Chili
- Salads
  - Caprese Salad*
  - Fleet Salad*
  - Cotton Salad
  - Kachumbari Salad*
- Pizzas
  - Firecracker Pizza
  - Quattro Formaggi Pizza
  - Pesto Pizza
  - Cotton Pizza
- Pastries
  - Cheesecake Balls*
  - Mothmallow + Slice*
  - Moffins

All dishes / ingredients have been added to the guidebook alongside a new Salad page to group all of those at the same place.

**Media**
![ss+(2023-03-15+at+06 09 09)](https://user-images.githubusercontent.com/9823203/225387245-758d9432-a35e-42cf-8441-f5c3012a90f2.png)

- [x] Add the other recipes related to moth except pizzas
- [x] "Balance" the food as the values are plucked directly from TG
- [x] Update PR with all the proper information.
- [x] Have Cabbage and Red Onion available.
- [x] Have Tofu and Tofu slice available.
- [x] Have Soybeans available.
- [x] Have Garlic available.
- [x] Add all the recipes to the guidebook

**Changelog**
<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged
There are 4 icons for changelog entries: add, remove, tweak, fix. I trust you can figure out the rest.

You can put your name after the :cl: symbol to change the name that shows in the changelog (otherwise it takes your GitHub username)
Like so: :cl: PJB

Generally, only put things in changelogs that players actually care about. Stuff like "Refactored X system, no changes should be visible" shouldn't be on a changelog.

For writing actual entries, don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers
-->

:cl:
- add: Added Mothic Cuisine with 3 new cheese, 3 new sauces and 41 recipes.

FluffiestFloof committed:
Moth cuisine init

Ported from TG moth cuisine.
Not finished.

FluffiestFloof committed:
Complex cotton flavor

FluffiestFloof committed:
More.

With curd cheese, cheese curds and mozzarella most definitely going to be made by someone upstream eventually.
But right now I need em so I made em.

FluffiestFloof committed:
Minor fixes

FluffiestFloof committed:
Minor balance

FluffiestFloof committed:
Nutriment

FluffiestFloof committed:
Forgot that they weren't a thing

This should do it.
Lots of placeholders but it works for now

FluffiestFloof committed:
No point in specific moth meat

Serves no real purpose but to reskin their meat.
Might be a fun idea but a bit pointless.

FluffiestFloof committed:
Random spawner and mail moth food

FluffiestFloof committed:
Moffin recipe change

FluffiestFloof committed:
Placeholders ingredients, remove medicines

Torun-S commented on Resources/Prototypes/Nyanotrasen/Recipes/Cooking/recipes_moth.yml:
```diff
@@ -0,0 +1,331 @@
+  id: RecipeMothButteredBakedCorn
+  name: buttered baked corn recipe
+  result: FoodMothButteredBakedCorn
+  time: 15
```
Just 5 secs should be enough to melt a bit of butter, let's be nice.

FluffiestFloof replied:
Reduced BakedCorn to 15s and ButteredBakedCorn to 5s so it takes 20 in total

FluffiestFloof committed:
Change Moffin recipe

Similar to pancake.

Torun-S commented on Resources/Prototypes/Nyanotrasen/Recipes/Reactions/food.yml:
```diff
@@ -0,0 +1,32 @@
+  impact: Low
+  quantized: true
+  reactants:
+    Milk:
+      amount: 15
+    Vinegar:
+      amount: 5
+    Cream:
+      amount: 5
+  effects:
+    - !type:CreateEntityReactionEffect
+      entity: FoodCurdCheese
```
20u worth of ingredients, but only 5u of food that comes out? That's immensely lossy, even for curdling.

FluffiestFloof replied:
Hmm you're right. Base CheeseWheel is 40u of ingredients for 15u of Food
So I could divide the ingredients requirement for Curd and Mozzarella so it roughly does 20u ingredients for 10u food

Torun-S commented on Resources/Prototypes/Nyanotrasen/Recipes/Reactions/food.yml:
```diff
@@ -0,0 +1,32 @@
+      catalyst: true
+  effects:
+    - !type:CreateEntityReactionEffect
+      entity: FoodMozzarella
```
Same question as with curd cheese.

Torun-S reviewed (commented):
Not saying you intended any of the below comments, you probably just ported, but still, if we're porting it, we might as well discuss and maybe tweak it!

FluffiestFloof committed:
Curdling makes more and Less corn time

FluffiestFloof committed:
Made Mozzarella 25 for 10 too

FluffiestFloof committed:
Clean up

FluffiestFloof committed:
Red Onions are in but not findable

:(

FluffiestFloof committed:
Upstream now has Cabbage and Red Onion

FluffiestFloof committed:
Replace Poppy with Ambrosia

Still placeholder but it fits with the other salads using those for herbs

FluffiestFloof committed:
Two more.

FluffiestFloof committed:
Three pizzas

Edited sprites by me, good enough for now

FluffiestFloof committed:
Baked Rice + Portion

FluffiestFloof committed:
Pizza changes

FluffiestFloof committed:
Pizza findable in box

FluffiestFloof committed:
Slight change

FluffiestFloof committed:
Merge branch 'master' into moffCuisine

FluffiestFloof committed:
Vegetarian Chili

Also change everything so it probably passes the YAML test until certain ingredients are added

FluffiestFloof committed:
Tofu, Garlic, Soybean, Vinaigrette

Upstreamed.

FluffiestFloof committed:
Fix some times

FluffiestFloof committed:
Pesto and Bechamel

FluffiestFloof committed:
TableSalt

FluffiestFloof committed:
Merge branch 'Nyanotrasen:master' into moffCuisine

FluffiestFloof committed:
Update guidebook for moth recipe

Plus changes from upstream

FluffiestFloof committed:
Merge branch 'Nyanotrasen:master' into moffCuisine

FluffiestFloof committed:
Fix

FluffiestFloof committed:
FoodSoybeans

Forgot a single letter :)

Torun-S commented on Resources/Server Info/Guidebook/Chef/salad.xml:
Valid Salad's missing, I think?

FluffiestFloof replied:
Somehow I lost it along the way.

Torun-S commented on Resources/Server Info/Guidebook/Chef/spaget.xml:
```diff
@@ -73,6 +78,22 @@
+</Box>
+<Box Orientation="Vertical">
+- Pesto
+- Bechamek
```
Typo!

Torun-S commented on Resources/Server Info/Guidebook/Chef/sup.xml:
```diff
@@ -113,4 +113,169 @@
+- 5u cream
+</Box>
+<Box Orientation="Vertical">
+- Tofu
```
Slice or block? It will matter since upstream has both.

FluffiestFloof committed:
Moth Eyeball Soup.

Just because someone asked for moth soup.

Torun-S commented on Resources/Prototypes/Nyanotrasen/Recipes/Cooking/meal_recipes.yml:
```diff
@@ -30,3 +30,11 @@
+  id: RecipeCheeseCurds
+  name: cheese curds recipe
+  result: FoodCheeseCurds
+  time: 20
```
20 seconds?

FluffiestFloof replied:
uhuh no idea, would 10 seconds work?

Torun-S replied:
Seeing as one piece of this cheese is one bite and that this is also an intermediate ingredient, 5 is more appropriate.

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/ingredients.yml:
Not sure there's a point in solution headways in ingredients (maxVol of solution is higher than the sum of its parts).

FluffiestFloof replied:
It's very inconsistent over multiple food items, some have higher maxvol, some have equal, some have lower(somehow). I'll just make it equals I guess.

Torun-S commented on Resources/Server Info/Guidebook/Chef/cookbook.xml:
We get any more Cheeses, we'll have to split them off into a new page. This is about fine for now, though.

Torun-S commented on Resources/Server Info/Guidebook/Chef/cookbook.xml:
```diff
@@ -121,4 +147,54 @@ Protein can be used raw, but sometimes has to be cooked beforehand. Simply micro
+</Box>
+<Box>
+<GuideEntityEmbed Entity="FoodCurdCheese" Caption=""/>
+<GuideEntityEmbed Entity="KitchenMicrowave" Caption="20 sec"/>
```
Again, 20 seconds?

Torun-S commented on Resources/Server Info/Guidebook/Chef/cookbook.xml:
```diff
@@ -121,4 +147,54 @@ Protein can be used raw, but sometimes has to be cooked beforehand. Simply micro
+- Ambrosia Vulgaris
+</Box>
+<GuideEntityEmbed Entity="KitchenMicrowave" Caption="5 sec"/>
+<GuideEntityEmbed Entity="FoodMothPesto" Caption=""/>
```
Suggest we normalize times and put 10s.

Torun-S commented on Resources/Server Info/Guidebook/Chef/cookbook.xml:
```diff
@@ -121,4 +147,54 @@ Protein can be used raw, but sometimes has to be cooked beforehand. Simply micro
+- Butter
+</Box>
+<GuideEntityEmbed Entity="KitchenMicrowave" Caption="5 sec"/>
+<GuideEntityEmbed Entity="FoodMothBechamel" Caption=""/>
```
Suggest we normalize times and put 10s.

Torun-S commented on Resources/Server Info/Guidebook/Chef/pitzah.xml:
```diff
@@ -96,4 +103,102 @@
+</Box>
+<Box Orientation="Vertical">
+- Dough
+- 2 Tomatoes
```
I'd be up for simplicity and merge the normal Margherita recipe with this one. Maybe get rid of using Cheddar and stick to mozzie, remove Ambrosia, and be done?

FluffiestFloof replied:
Would using the current amount of ingredients for the current Margherita work? 1 Mozzarella, 1 Dough, 4 Tomatoes.

Torun-S replied:
Hmm, 4 tomatoes is steep, I'm thinking 3 tomato and 2 mozzarella, it lines up better with typical harvests and mozzarella is made in pairs anyway.

Torun-S commented on Resources/Server Info/Guidebook/Chef/pitzah.xml:
```diff
@@ -96,4 +103,102 @@
+</Box>
+
+<Box Orientation="Vertical">
+	[color=#a4885c]Mothic Five-Cheese Pizza[/color]
```
It's only got four cheeses in it. 3.5, really since curd cheese and cheese curd is still two pieces of curdy cheese where one was heated in a microwave.

Torun-S commented on Resources/Server Info/Guidebook/Chef/pitzah.xml:
```diff
@@ -96,4 +103,102 @@
+<Box Orientation="Vertical">
+- Garlic
+- Ambrosia Vulgaris
+</Box>
```
These two recipes might do with a merging that would result in a pizza equivalent of Aglio i Olio. Basically a Garlic Pizza Bread. In fact, toppings of those two are pretty much exactly a formula of Aglio i Olio split in two, and why keep garlic and pesto from each other is beyond me. And "light" pizzas like that are a perfectly valid food.

FluffiestFloof replied:
Eh, I'd argue a Pesto Pizza and Garlic Pizza have different flavors.
I'd make the Pesto recipe use tomatoes and the Garlic use Mozzarella instead of Herbs to differentiate them a bit if that can work.

Torun-S replied:
See, Pesto is already everything the Garlic Pizza is because it's made with herbs, garlic, oil. At best, you can make the Garlic Pizza an alternate recipe, but they're both nearly identical in contents, except one's made with olivol and the other with butter. Normally, I'd be up for variety in preparing foods, except the cooking system is shit at it, so I lean on the side of debloating, and it's unnecessary here because olivol isn't hard to come by, so you don't need another recipe as a substitute.

Torun-S commented on Resources/Server Info/Guidebook/Chef/salad.xml:
Isn't Valid Salad missing?

Torun-S commented on Resources/Server Info/Guidebook/Chef/salad.xml:
```diff
@@ -0,0 +1,151 @@
+</Box>
+<Box Orientation="Vertical">
+- Slice of bread
+- Cheese wedge
```
Wouldn't mozzarella fit Caesar salad more?

FluffiestFloof replied:
Caesar salad was made for upstream so I could get my cabbage so why not if it makes more sense

Torun-S commented on Resources/Server Info/Guidebook/Chef/snacc.xml:
```diff
@@ -112,80 +121,124 @@
 </Box>
 <Box>
+<GuideEntityEmbed Entity="FoodCheese" Caption=""/>
+<GuideEntityEmbed Entity="KitchenMicrowave" Caption="20 sec"/>
```
Normalize cooking time, 15s

Torun-S commented on Resources/Prototypes/Nyanotrasen/Recipes/Cooking/recipes_moth.yml:
```diff
@@ -0,0 +1,496 @@
+  time: 5
+  solids:
+    FoodMothBakedCorn: 1
+    FoodButter: 1
```
Making a note that one of my refactor targets is using molten butter in places, and this is a prime example; A whole stick for one ear of corn could use a pass of common sense.

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        - ReagentId: Nutriment
+          Quantity: 2
+        - ReagentId: Vitamin
+          Quantity: 4
```
Healthy food is one thing, but it should also be somewhat nutritious. I know it makes sense for salads, but when players catch on that they don't quench hunger, they might not bother making it.

FluffiestFloof replied:
I think the original idea for it was to serve as a base for other salads so I kept it to a similar status. Will boost it to 4nutri 2vit

Torun-S replied:
Cool!

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        maxVol: 5
+        reagents:
+        - ReagentId: Vitamin
+          Quantity: 3
```
Remember to add Allicin or whatever the reagent in Garlic is going to be. That goes for **every** food with garlic in it.

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        maxVol: 5
+        reagents:
+        - ReagentId: Vitamin
+          Quantity: 3
```
Bechamel is among the most decadent and fatty things out there, it's an instant heart attack eaten alone. Also, 25u liquids + flour into 3u of reagents is not great. I would probably make it contain Cream and Oil and a touch of Nutriment, 20u total. And given it's pretty much always an intermediate ingredient (since it's a sauce), I'd rather see it as a liquid, but that's something for later.

FluffiestFloof replied:
I could solve that by creating a condiment/reagent for the Bechamel/Pesto and have the microwave recipe create a bowl filled with 20u of said reagent.
Then make the recipes use only 10u of those sauce so it makes it less costly overall.

Torun-S replied:
Awesome.

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+  - type: SolutionContainerManager
+    solutions:
+      food:
+        maxVol: 20
```
maxVol to 25 for condiments

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        - ReagentId: Protein
+          Quantity: 6
+        - ReagentId: Vitamin
+          Quantity: 12
```
Take 4 Vitamin and put to Nutriment

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        - ReagentId: Protein
+          Quantity: 1
+        - ReagentId: Vitamin
+          Quantity: 2
```
0.7

FluffiestFloof replied:
changed it to 1.25 because 8 divided in 6 'slice' will result in 1.33 vitamin per slice

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        - ReagentId: Nutriment
+          Quantity: 24
+        - ReagentId: Protein
+          Quantity: 6
```
Change 6 Nutriment to Vitamin

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        - ReagentId: Nutriment
+          Quantity: 3
+        - ReagentId: Protein
+          Quantity: 1
```
Add 1 maxVol and 1 Vitamin

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        maxVol: 14
+        reagents:
+        - ReagentId: Nutriment
+          Quantity: 12
```
8 Nutriment

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+          Quantity: 12
+#Tastes like cornmeal, cheese, eggplant and tomato sauce
+
+- type: entity
+  name: vegetarian chili
+  parent: FoodBowlBase
+  id: FoodMothVegetarianChili
+  description: For the hombres who don't want carne.
```
"Vegan Chili. Made of real vegans."

Rosy's idea, and as always, I vibe with it.

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        - ReagentId: Protein
+          Quantity: 4
+        - ReagentId: Sugar
+          Quantity: 12
```
Max 15, 2 Nutri, 1 Protein, 12 Sugar.

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -0,0 +1,1210 @@
+        - ReagentId: Nutriment
+          Quantity: 20
+        - ReagentId: Sugar
+          Quantity: 10
```
20 maxVol, 50/50 Nutri and Sugar

Torun-S reviewed (changes requested):
28 comments... Careful about asking me to review anything.

There's a lot of Ambrosia Vulgaris in a lot of recipes, used as a kind of a stand-in for herbs common in Mediterranean cuisine like Rosemary, Thyme, Oregano, Basil... And Botany doesn't often grow it. We might want to consider curbing that a little, but I have way too many other ideas to mull over to care right now.

I tried looking into nutrients of the foods to make them make sense in accordance to the nature of the dish and the volumes going in and out, and also found some recipes which might not actually need to exist, and many other things.

Fun read!

FluffiestFloof committed:
Update after review

- Made Pesto and Bechamel into reagents. Pesto is OliveOil, Salt, Allicin, Nutriment. Bechamel Milk, Flour, Butter
- Made use of grinded/melted butter in all recipes
- Merged the two Margherita Pizza recipe into one for simplicity
- Removed the Garlic Pizza and changed the Pesto Pizza recipe
- Fixed maxVol being incorrect for a bunch of stuff
- Fixed some reagents quantities being incorrect
- Renamed the pizza as the Mothic was being useless
- Changed some cooking times
- Fix typos
- Modified some recipes to use Mozzarella
- Re-added the valid salad :)
- Updated the macballs sprite so the small sauce bowl isn't horrible
- Added a condiment section for the guidebook as some have recipes.

FluffiestFloof committed:
additional text

Torun-S commented on Resources/Server Info/Guidebook/Chef/sos.xml:
I am happy you follow the naming convention diligently!

In the recipe output where the caption is like "Xu of Y", just list the units, the "header" of a recipe should make it clear enough what you're making.

Torun-S commented on Resources/Server Info/Guidebook/Chef/snacc.xml:
```diff
@@ -122,6 +122,27 @@
+- Curd Cheese
+- Ambrosia Vulgaris
+</Box>
+<GuideEntityEmbed Entity="KitchenMicrowave" Caption="10 sec"/>
```
Normalize. 5 sec, or 15, but not 10. Leaning to 5 since it's still one smol item with one herb.

Torun-S commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Consumable/Food/moth.yml:
```diff
@@ -46,14 +46,76 @@
+    sprite: Nyanotrasen/Objects/Consumable/Food/moth.rsi
+    state: condimentbowl
+  - type: Icon
+    sprite: Nyanotrasen/Objects/Consumable/Food/moth.rsi
+    state: condimentbowl
+  - type: TrashOnEmpty
+    solution: food
```
Well done, but the trash bit makes me wonder... What if we required a smol bowl or another smaller vessel in the recipes for this and then changed the trash item to that same vessel? Things like steaks used to work like that, pies and spaget still does. We'd only need to change the base entity for the trash part, but it would still mean changing all recipes that use this base because there's no parent recipes... so far.

FluffiestFloof replied:
Had it to be like that when I used the soup as a base, it kinda worked but to have the entity itself go from "Soup" to "Empty Bowl" I had to try to drink from it, only then it would be like "oh ye im empty" and become a normal bowl again.
Could potentially re add that if the trash part is really an issue.

Torun-S replied:
Oh, I see, yeah, that feels kind of hacky and unreliable, just have it be a single use thing. Don't want to risk someone losing bowls to this.

Torun-S commented on Resources/Prototypes/Guidebook/chef.yml:
```diff
@@ -10,10 +10,11 @@
+  - CookBurger
+  - CookSauce
   - CookSandwich
   - CookSnack
```
Place Snaccs on top of the list, it's our catch-all and I would like to lower the chance of it being dismissed. Don't have a better name for it yet, though.

Torun-S replied:
I did mean snacks.

Torun-S commented on Resources/Server Info/Guidebook/Chef/sup.xml:
```diff
@@ -238,7 +238,7 @@
 </Box>
 <Box Orientation="Vertical">
-- Tofu
+- Tofu block
```
For the kind of nutrition you get out of heartburn soup, one slice is adequate!

Torun-S reviewed (commented):
Nitpicking at this point, and also took another look at nutrition stats after the recipe changes to make sure it still adds up.

FluffiestFloof committed:
Additional fixes

- Removes unnecessary name after units
- Reduce herby cheese to 5 second, it's literally just cheese and herbs anyway
- Places back snacks at the top
- Tofu block to Tofu slice for heartburn

FluffiestFloof committed:
Actually place back snack at the top

Torun-S reviewed (approved):
I'll leave you alone now :D Well done!
